packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWDrawer/CWDrawer.tsx

```diff
@@ -0,0 +1,62 @@
+import useBrowserWindow from 'client/scripts/hooks/useBrowserWindow';
+import React, { ComponentProps, useState } from 'react';
+import Drawer from 'react-modern-drawer';
+import 'react-modern-drawer/dist/index.css';
+import { CWIcon } from '../../cw_icons/cw_icon';
+import { CWText } from '../../cw_text';
+import { ComponentType } from '../../types';
+import './CWDrawer.scss';
+
+type CWDrawerProps = Omit<ComponentProps<typeof Drawer>, 'direction'> & {
+  direction?: 'left' | 'right' | 'top' | 'bottom';
+  header: string;
+};
+
+export const CWDrawer = ({
+  direction,
+  duration,
+  open,
+  onClose,
+  children,
+  header,
+}: CWDrawerProps) => {
+  const { isWindowExtraSmall } = useBrowserWindow({});
+  const [fullscreen, setFullscreen] = useState(false);
+
+  const toggleDrawerFullscreen = () => {
+    setFullscreen((prevState) => !prevState);
+  };
+
+  return (
+    <Drawer
+      duration={duration}
+      size={fullscreen ? '100vw' : isWindowExtraSmall ? '90vw' : '50vw'}
+      open={open}
+      onClose={onClose}
+      direction={direction || 'right'}
+      enableOverlay={false}
+      className={ComponentType.Drawer}
+      // moves the drawer below the top navigation bar
+      style={{ top: '56px' }}
+    >
+      <div className="drawer-actions">
+        <CWIcon
+          iconName="caretDoubleRight"
+          onClick={onClose}
+          iconSize="small"
+        />
+        {!isWindowExtraSmall && (
```

Comment:
My only comment (besides echoing Marcin's's `cursor: pointer` catch) is that, in the mobile designs I see in #6005, Mobile drawers retain the ability to expand to full screen? Perhaps this is out of date, but flagging

Reply:
Will comment below on fullscreen functionality!